hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -388,13 +388,13 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
             currentInstantLogBlocks = new ArrayDeque<>();
             dedupedLogBlocksInfo.getValue().forEach(block -> currentInstantLogBlocks.push(block));
             LOG.info("Merging the final data blocks");
-            processQueuedBlocksForInstant(currentInstantLogBlocks, scannedLogFiles.size(), keySpecOpt);
+            processQueuedBlocksForInstant(currentInstantLogBlocks, scannedLogFiles.size(), keySpecOpt, logFormatReaderWrapper);
```

Comment:
So there is no way to reduce the maximum memory for all these log blocks.

Reply:
I guess one way would be to port the design change of 1.0 to 0.14.1 (i.e. remove the log file appending for multiple instants). But, I am little skeptical of doing that in a patch release.
Or else, close the input stream immediately after scanning header and next block offset.

Reply:
We had an offline discussion, the input stream should be closed immediately after the data block header is fetched, we should re-create the input stream again with the correct offset when deserializing the records from content. That many incrrease several rpc to data storage, but in object storage it is not a bottleneck.